### PR TITLE
Increment execution timings on errors as well

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
When execution timings spike, in `replay-slot-stats`, none of the constituent execution details timings go up the proportional amount. 

One contributor to this could be that early returns for execution failures aren't being counted towards the execution details timings

#### Summary of Changes
Increment execution details timings even when failures occur

Fixes #

### CODE REVIEW

File: programs/bpf_loader/src/lib.rs [hunk 1]
```diff
@@ -995,7 +995,7 @@ impl Executor for BpfExecutor {
         serialize_time.stop();
         let mut create_vm_time = Measure::start("create_vm");
         let mut execute_time;
-        {
+        let execuction_result = {
```
Comment (sakridge) [inline programs/bpf_loader/src/lib.rs:998]: `execution_result`

File: programs/bpf_loader/src/lib.rs [hunk 2]
```diff
@@ -1058,30 +1056,38 @@ impl Executor for BpfExecutor {
...
         let timings = &mut invoke_context.timings;
         timings.serialize_us = timings.serialize_us.saturating_add(serialize_time.as_us());
         timings.create_vm_us = timings.create_vm_us.saturating_add(create_vm_time.as_us());
         timings.execute_us = timings.execute_us.saturating_add(execute_time.as_us());
         timings.deserialize_us = timings
             .deserialize_us
             .saturating_add(deserialize_time.as_us());
+
+        execute_or_deserialize_result?;
         stable_log::program_success(&log_collector, &program_id);
```
Comment (t-nelson) [inline programs/bpf_loader/src/lib.rs:1091]:
```suggestion
        if execute_or_deserialize_result.is_ok() {
            stable_log::program_success(&log_collector, &program_id);
        }
```

Might make future modifications in this area a little less error prone than the early return

**Review by sakridge — DISMISSED** [review]
good find

**Review by t-nelson — APPROVED** [review]
lgtm!  Just one nit/suggestion

File: programs/bpf_loader/src/lib.rs (hunk 2 above)
Comment (jackcmay) [reply programs/bpf_loader/src/lib.rs:1091]: would still need to return an error here, the suggestion above always returns `Ok(())` if I'm reading that right

**Review by jackcmay — DISMISSED** [review]
lgtm + my comment

**Comment by carllin** [comment]
@taozhu-chicago quick question on the similar issue here: https://github.com/solana-labs/solana/blob/9c5d82557a965cf23831a3fb2e3b8986d379ce04/runtime/src/message_processor.rs#L115-L123

The call to `accumulate_program()` needs a `compute_meter_consumption`, which will not be available if the `process_instruction` call errors https://github.com/solana-labs/solana/blob/9c5d82557a965cf23831a3fb2e3b8986d379ce04/runtime/src/message_processor.rs#L115-L117

I'm assuming we still want to `accumulate_program` even on errors, so is the appropriate thing to do to change `process_instruction: https://github.com/solana-labs/solana/blob/9c5d82557a965cf23831a3fb2e3b8986d379ce04/program-runtime/src/invoke_context.rs#L653-L673 to always return the consumed compute, even on error?

**Comment by taozhu-chicago** [comment]
> @taozhu-chicago quick question on the similar issue here:
> 
> https://github.com/solana-labs/solana/blob/9c5d82557a965cf23831a3fb2e3b8986d379ce04/runtime/src/message_processor.rs#L115-L123
> 
> The call to `accumulate_program()` needs a `compute_meter_consumption`, which will not be available if the `process_instruction` call errors
> 
> https://github.com/solana-labs/solana/blob/9c5d82557a965cf23831a3fb2e3b8986d379ce04/runtime/src/message_processor.rs#L115-L117
> 
> I'm assuming we still want to `accumulate_program` even on errors, so is the appropriate thing to do to change `process_instruction:
> 
> https://github.com/solana-labs/solana/blob/9c5d82557a965cf23831a3fb2e3b8986d379ce04/program-runtime/src/invoke_context.rs#L653-L673
> 
> to always return the consumed compute, even on error?

Agree that CU and exexute_us should always be aggregated and reported for each execution, regardless of success or error.

**Review by taozhu-chicago — DISMISSED** [review]
lgtm

File: programs/bpf_loader/src/lib.rs (hunk 2 above)
Comment (t-nelson) [reply programs/bpf_loader/src/lib.rs:1091]: Of course can't fix the recommendation without deleting the comment :man_facepalming: 

```
-        execute_or_deserialize_result?;
-        stable_log::program_success(&log_collector, &program_id);
-        Ok(())
+        if execute_or_deserialize_result.is_ok() {
+            stable_log::program_success(&log_collector, &program_id);
+        }
+        execute_or_deserialize_result
```

**Pull request merged**